### PR TITLE
Fixed program admin bug

### PR DESCRIPTION
**Pull request by clintonb**
The previous override of save_model() was largely unnecessary and wrong. The override now relies on the super method, and simply does error handling.

ECOM-6238

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review. This solves the immediate problem. Long-term, we should investigate options around using a wizard. At the very least, on publisher errors, we should direct the user back to the form rather than show the error page.

File: course_discovery/apps/course_metadata/forms.py [hunk 1]
```diff
@@ -16,7 +16,6 @@ class HackDjangoAutocompleteMixin(object):
     # fact an issue?
 
     class QuerySetSelectMixin2(widgets.WidgetMixin):
-
```
Comment (mikedikan) [inline course_discovery/apps/course_metadata/forms.py:19]: curious why this wasn't a quality error before?

File: course_discovery/conf/locale/eo/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -237,6 +237,24 @@ msgstr "Pärtnér Ⱡ'σяєм ιρѕυм #"
...
+#: apps/course_metadata/admin.py
+msgid "Included course runs"
+msgstr "Ìnçlüdéd çöürsé rüns Ⱡ'σяєм ιρѕυм ∂σłσя ѕιт αмєт, #"
+
+#: apps/course_metadata/admin.py
+msgid ""
+"An error occurred while publishing the program to the marketing site. Please"
+" try again. If the error perisists, please contact the Engineering Team."
+msgstr ""
+"Àn érrör öççürréd whïlé püßlïshïng thé prögräm tö thé märkétïng sïté. Pléäsé"
```
Comment (mikedikan) [inline course_discovery/conf/locale/eo/LC_MESSAGES/django.po:249]: what is this?

**Comment by awais786** [comment]
There are some test cases for this admin page. I am surprised why they did not fail due to the bug.

**Comment by MatthewPiatetsky** [comment]
@clintonb have you tested that the drag and drop widget functionality still works in the sense that after you drag and drop and save when you reload the new order is preserved?

File: course_discovery/apps/course_metadata/forms.py (hunk 1 above)
Comment (clintonb) [reply course_discovery/apps/course_metadata/forms.py:19]: It's not a quality error. Having this newline here is more of a personal choice. In my case, I rely on PyCharm's code cleanup tools.

File: course_discovery/conf/locale/eo/LC_MESSAGES/django.po (hunk 2 above)
Comment (clintonb) [reply course_discovery/conf/locale/eo/LC_MESSAGES/django.po:249]: Dummy strings generated to validate that all strings are marked for translation. See http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/conventions/internationalization/i18n.html.

**Review by schenedx — APPROVED** [review]
👍

**Comment by MatthewPiatetsky** [comment]
@clintonb I just tried this locally, and I think the drag and drop no longer works due to the save_model change (the purpose of using the cleaned data was to get the data in the new order post drag and drop).  
You'll notice if you drag and drop, save, once the page reloads, you're back to the original order before the drag and drop.

**Comment by clintonb** [comment]
Drag and drop works, and maintains the order. The issue is the JS widget.

**Comment by MatthewPiatetsky** [comment]
yes you can physically drag and drop on the screen, but that new order does not get saved to the model. 
the purpose of that save_model change was to save the new order.
If that change prevents creating new programs we should fix that, but we shouldn't break the ability to order courses and organizations. (This will be a regression for several completed tickets)

**Comment by clintonb** [comment]
As I mentioned, order is maintained. Django forms behave correctly. The issue is that your JS widget displays the items alphabetically. I am investigating.

**Comment by clintonb** [comment]
Replaced by #409.

**Pull request closed without merging**